src/components/taskItem/TaskItem.tsx

```diff
@@ -0,0 +1,40 @@
+import styles from "./TaskItem.module.css";
+import { Dispatch, ReactElement, SetStateAction } from "react";
+import { Task } from "../../types/Task.types";
+import { deleteTask, updateTask } from "../../utility/api";
+
+export type Props = Task & {
+  setTasks: Dispatch<SetStateAction<Task[]>>;
```

Comment:
That makes sense. I updated it but it caused a ton of other issues that I'd need to work through. Since my next goal is to implement the app with redux, it won't be necessary for me to pass the setTasks() function from the parent component to the child components, and this change will be unnecessary. What do you think about leaving it as is for now since this property + type will be removed after the redux implementation anyway?

Reply:
What kind of problem do you have?
I rewrote your app with no issues.

Reply:
Left a comment on your PR, everything makes sense now, updating!